### PR TITLE
Enhanced tests

### PR DESCRIPTION
**Pull request by rtrvrtg**
Updated tests and fixed a few registration issues.

### CODE REVIEW

**Comment by rtrvrtg** [comment]
Ran this through hlint with hints applied.

File: lib/Data/OpenSRS.hs [hunk 1]
```diff
@@ -115,50 +120,59 @@ doRequest' parser r = do
...
+                   (read $ getText xml "<item key='response_code'>")
 
 --------------------------------------------------------------------------------
 -- | Parse domain list
 parseDomainList :: String -> DomainList
-parseDomainList s = DomainList (read $ gt "<item key='total'>")
-                               (Prelude.map makeDomain domainItems)
-                               (gt "<item key='remainder'>" == "1")
+parseDomainList s = DomainList (read $ getText xml "<item key='total'>")
+                               (fmap makeDomain domainItems)
```
Comment (christian-marie) [inline lib/Data/OpenSRS.hs]: Much nicer

File: lib/Data/OpenSRS.hs [hunk 2]
```diff
@@ -115,50 +120,59 @@ doRequest' parser r = do
...
+                   (getText xml "<item key='response_text'>")
+                   (read $ getText xml "<item key='response_code'>")
 
 --------------------------------------------------------------------------------
 -- | Parse domain list
 parseDomainList :: String -> DomainList
-parseDomainList s = DomainList (read $ gt "<item key='total'>")
-                               (Prelude.map makeDomain domainItems)
-                               (gt "<item key='remainder'>" == "1")
+parseDomainList s = DomainList (read $ getText xml "<item key='total'>")
```
Comment (christian-marie) [inline lib/Data/OpenSRS.hs]: Read is never a good idea, not sure if it's 100% safe here, but read in production code is almost always bad.

As long as you have base >= 4.6, try to use readMaybe

File: lib/Data/OpenSRS/ToXML.hs [hunk 3]
```diff
@@ -14,122 +16,121 @@ import Text.XmlHtml
...
 domainToNodes :: Domain -> [Node]
 domainToNodes domain = [
     itemNode "domain" (domainName domain),
     itemParent "contact_set" [
-        tag "dt_assoc" (concatMap contactToNodes $ toList $ domainContactSet domain)
+        tag "dt_assoc" ((=<<) contactToNodes . toList $ domainContactSet domain)
     ],
     itemParent "nameserver_list" [
-        tag "dt_array" (concatMap nsToNodes $ zip [0..] $ domainNameservers domain)
+        tag "dt_array" ((=<<) nsToNodes . zip [0..] $ domainNameservers domain)
```
Comment (christian-marie) [inline lib/Data/OpenSRS/ToXML.hs]: (=<<) is fine here, but, it's meant to be placed infix. Hlint is a bit dumb when it suggests to use it as (=<<)

**Pull request merged**